frontend/src/utils/fetchData.tsx

```diff
@@ -16,7 +18,20 @@ const nonNeededToken = ['/auth/login', '/auth/refresh', '/auth/registerAzure'];
 
 export const getToken = async (context?: GetServerSidePropsContext) => {
   const session = await getSession(context);
-  if (session) return `Bearer ${session?.user.accessToken.token}`;
+
+  // accessing cookies depends on where it is called (server side or client side)
+  let guestUserCookie: GuestUser;
+  if (context) {
+    const { req, res } = context;
+    guestUserCookie = getGuestUserCookies({ req, res }, true);
+  } else guestUserCookie = getGuestUserCookies();
```

Comment:
```suggestion
if (!session) {
  if (context) {
    const { req, res } = context;
    guestUserCookie = getGuestUserCookies({ req, res }, true);
  } else guestUserCookie = getGuestUserCookies();
}
```

This shouldn't  be executed if the user is logged in